nipoppy/extractors/networks/extract_network.py

```diff
@@ -0,0 +1,355 @@
+    streamlines = trk.streamlines
+
+    print("Creating adjacency matrices for found dseg labels...")
+
+    # loop found parcellations and create networks
+    for parc in parcs:
+
+        print('Loading parcellation:', parc)
+        labels_file = nib.load(os.path.join(inp_labels, parc))
+        labels = labels_file.get_fdata()
+
+        np.unique(labels)
+
+        print('Building the adjacency matrix with the loaded tractography and labels...')
+        M, grouping = utils.connectivity_matrix(streamlines, affine, labels.astype(np.uint16),
+                                                return_mapping=True, mapping_as_streamlines=True)
+
+        # drop the first row / column (label 0 is background)
+        M = M[1:, 1:]
+
+        # create output name
+        outf = os.path.join(inp_output, parc.replace('_dseg.nii.gz', '_conmat-count.tsv'))
+
+        print('Saving the structural connectome:', outf)
+        np.savetxt(outf, M, delimiter='\t')
+
+
+def run(participant_id, global_configs, session_id, output_dir,
+        tractoflow_dir, fmriprep_dir, freesurfer_dir, logger=None):
+    """
+    Parse the inputs to create the DKT and Schaefer networks for the requested participant
+    """
+
+    # copy fmriprep config parsing - minor extensions for tractoflow needed
+
+    # extract paths
+    DATASET_ROOT = global_configs["DATASET_ROOT"]
+    TEMPLATEFLOW_DIR = global_configs["TEMPLATEFLOW_DIR"]
+    CONTAINER_STORE = global_configs["CONTAINER_STORE"]
+    FMRIPREP_CONTAINER = global_configs["PROC_PIPELINES"]["fmriprep"]["CONTAINER"]
+    TRACTOFLOW_CONTAINER = global_configs["PROC_PIPELINES"]["tractoflow"]["CONTAINER"]
+    FMRIPREP_VERSION = global_configs["PROC_PIPELINES"]["fmriprep"]["VERSION"]
+    TRACTOFLOW_VERSION = global_configs["PROC_PIPELINES"]["tractoflow"]["VERSION"]
+    FS_VERSION = global_configs["PROC_PIPELINES"]["freesurfer"]["VERSION"]
+    FMRIPREP_CONTAINER = FMRIPREP_CONTAINER.format(FMRIPREP_VERSION)
+    TRACTOFLOW_CONTAINER = TRACTOFLOW_CONTAINER.format(TRACTOFLOW_VERSION)
+
+    # pick the container - I'm not sure fMRIPrep has Dipy - TractoFlow container definitely has ANTs / Dipy
+    SINGULARITY_FMRIPREP = f"{CONTAINER_STORE}{FMRIPREP_CONTAINER}"
+    SINGULARITY_TRACTOFLOW = f"{CONTAINER_STORE}{TRACTOFLOW_CONTAINER}"
+
+    # set up logs
+    #log_dir = f"{DATASET_ROOT}/scratch/logs"
+
+    #if logger is None:
+    #    log_file = f"{log_dir}/network_extractor.log"
+    #    logger = my_logger.get_logger(log_file)
+
+    #logger.info("-"*75)
+    #logger.info(f"Using DATASET_ROOT: {DATASET_ROOT}")
+    #logger.info(f"Using participant_id: {participant_id}, session_id:{session_id}")
+    # logger.info(f"Optional args: --anat_only={anat_only}, --use_bids_filter={use_bids_filter}")
+    print(f"Using DATASET_ROOT: {DATASET_ROOT}")
+    print(f"Using PARTICIPANT_ID: {participant_id}, SESSION_ID: {session_id}")
+
+    if output_dir is None:
+        output_dir = f"{DATASET_ROOT}/derivatives/networks/v0.9.0"
```

Comment:
Should the `output_dir` be named with same version as the tractoflow/nextflow version that was run - should be easy to grab from the argparse flags?